integration/spark/src/main/scala/org/apache/spark/sql/execution/command/management/CommonLoadUtils.scala

```diff
@@ -767,7 +773,16 @@ object CommonLoadUtils {
             internalRow.getInt(index) + DateDirectDictionaryGenerator.cutOffDate)
         }
       }
-      internalRow
+      if (isInsertFromStageCommand) {
+        // Insert stage command, logical plan already consist of LogicalRDD of internalRow.
+        // When it is converted to DataFrame, spark is reusing the same internalRow.
+        // So, need to have a copy before the last transformation.
+        // TODO: Even though copying internalRow is faster, we should avoid it
+        //  by finding a better way
+        internalRow.copy()
+      } else {
+        internalRow
+      }
```

Comment:
How about moving code block to line 750?
Because this change faced NullPointerException in some scenarios.

Reply:
@QiangCai: CI all test case run. when we can get NPE ?
Shall I just add null check in the same line 782 ?  Becuase moving to line 750, how it can avoid NPE ?

Reply:
ok. Moved.